### ISSUE TITLE
optimize_minimal_flux should use the new solver API

### ISSUE THREAD

**Issue by aebrahim** [opencobra/cobrapy#228]
By creating a single LP object and using that to solve the problem initially (to find the optimal biomass flux), and using that same LP to compute the minimized sum of flux state, the basis can be reused which will make the second computation faster and more efficient.

If an already irreversible model is passed in, this might allow parsimonious FBA to actually be fast.

**Comment by cdiener** [comment]
Would love to help out with that since this is the cobra feature I use most. I guess by "new" solver API you refer to the strategy that is used in variability.py?

**Comment by aebrahim** [comment]
Thanks, help would be very appreciated, as I've been too busy to implement new features lately.

Yup, exactly the same strategy as variability. Basically, once the model is irreversible, create an LP, solve it, then modify that lp to make it a total flux minimization, and solve it again. Resuing the same LP will be a lot faster.

Also, @pstjohn mentioned he was also working on adding a fraction of optimum feature to the function. In case you want to coordinate with him.

**Comment by cdiener** [comment]
Yes we should definitely coordinate then.

**Comment by pstjohn** [comment]
I haven't had the chance to look into implementing this, although you're right that it would be faster to re-use the same basis. Give me a couple hours to see if its as easy as I expect though, I already have a re-write of the parsimonious implementation in #227 

Some quick profiling I just did on the existing methods:
```python
model = cobra.test.create_test_model('textbook')
%timeit cobra.manipulation.convert_to_irreversible(model.copy())
100 loops, best of 3: 3.07 ms per loop

%timeit model.optimize()
100 loops, best of 3: 7.31 ms per loop

%timeit optimize_minimal_flux(model)
100 loops, best of 3: 18.6 ms per loop

model = cobra.test.create_test_model('ecoli')
%timeit cobra.manipulation.convert_to_irreversible(model.copy())
10 loops, best of 3: 69.1 ms per loop

%timeit model.optimize()
10 loops, best of 3: 129 ms per loop

%timeit optimize_minimal_flux(model)
1 loops, best of 3: 465 ms per loop
```

It is theoretically possible to solve an absolute value minimization problem using linear programming directly, you just have to create dummy variables x_pos and x_neg and impose the right constraints. But that probably wouldn't allow us to re-use the same basis, and the current solver interface might make that more trouble than its worth.

**Comment by aebrahim** [comment]
When the model is irreversible, it could be done a lot more simply actually
with the same LP without using any dummy variables. Just fix the objective
flux bounds, set all objective coefficients to 1 and minimize.

**Comment by cdiener** [comment]
Looks like @pstjohn is already on it so I'll sit back.

I also think it will be much faster. Just compare to running FVA which runs a lot of optimizations in a fractionof time. In that sense wouldn't it be feasible to split up the second optimization into its own function as it is done in variability.py? This would allow to pass a solver and the objective value so one can iterate fast over many pFBAs for the same model. For instance assume I want to get the minimum total flux for a set of fractions of the original objective such as [0.5, ..., 0.9, 1].

I think that removing the original the original objective coefficients from the second optimization does not yield any advantage. The branching in the for loop is probably slower than the reduction step of the solver which removes them anyways. So simply setting all objective coefficients to 1 in the irreversible model is probably sufficient.

**Comment by aebrahim** [comment]
Fixed in #227
